OneSignalSDK/onesignal/notifications/src/main/java/com/onesignal/notifications/activities/NotificationOpenedActivityBase.kt

```diff
@@ -44,18 +45,26 @@ abstract class NotificationOpenedActivityBase : Activity() {
         processIntent()
     }
 
-    private fun processIntent() {
+    internal open fun processIntent() {
         if (!OneSignal.initWithContext(applicationContext)) {
             return
         }
-        suspendifyOnThread {
-            val openedProcessor = OneSignal.getService<INotificationOpenedProcessor>()
-            openedProcessor.processFromContext(this, intent)
-            // KEEP: Xiaomi Compatibility:
-            // Must keep this Activity alive while trampolining, that is
-            // startActivity() must be called BEFORE finish(), otherwise
-            // the app is never foregrounded.
-            finish()
-        }
+        suspendifyOnThread(
+            block = {
+                val openedProcessor = OneSignal.getService<INotificationOpenedProcessor>()
+                openedProcessor.processFromContext(this, intent)
+                // KEEP: Xiaomi Compatibility:
+                // Must keep this Activity alive while trampolining, that is
+                // startActivity() must be called BEFORE finish(), otherwise
+                // the app is never foregrounded.
+            },
+            onComplete = {
+                // Safely finish the activity on the main thread after processing is complete.
+                // This gives the system enough time to complete rendering before closing the Trampoline activity.
+                runOnUiThread {
+                    AndroidUtils.finishSafely(this)
+                }
+            },
```

Comment:
Why add this new `onComplete` instead of just adding this to the existing `block` above? This adds complexity that doesn't' seems to be needed.

Reply:
Thinking about it again, the pattern of `onComplete` make sense to split up code responsibilities. The `block` being the main work we want to do, and the `onComplete` to "clean up state" that isn't directly related to the code in the `block`.

If the above is correct can you add parameter comments for both  `block` and  `onComplete` parameters in `ThreadUtils.kt` to explain the recommended usage pattern?